libsignal-protocol/src/keys/identity_key_pair.rs

```diff
@@ -30,17 +33,36 @@ impl IdentityKeyPair {
         }
     }
 
-    pub fn public(&self) -> Result<PublicKey, Error> {
+    pub fn public(&self) -> PublicKey {
         unsafe {
             let raw = sys::ratchet_identity_key_pair_get_public(
                 self.raw.as_const_ptr(),
             );
             assert!(!raw.is_null());
-            Ok(PublicKey {
+            PublicKey {
                 raw: Raw::copied_from(raw),
-            })
+            }
+        }
+    }
+
+    pub fn private(&self) -> PrivateKey {
```

Comment:
i think you forgot to implement that method, so i added it for you xD

Reply:
Hehe, oops! Thanks for picking that up.